types/src/key.rs

```diff
@@ -345,7 +345,7 @@ impl Key {
         }
     }
 
-    /// Returns a reference to the inner [`DictionaryAddr`] if `self` is of type
+    /// Returns a reference to the inner `DictionaryAddr` if `self` is of type
```

Comment:
This has actually made me look at why the doc warning was saying the type isn't public.

Here, I think we should leave the doc comment as it was, but instead in lib.rs we should be publicly exporting `KEY_DICTIONARY_LENGTH` and `DictionaryAddr`.

Also, the doc comment for `DictionaryAddr` should be changed from ``/// A newtype for [`Key`]s dictionary variant.`` to ``/// An alias for [`Key`]s dictionary variant.``